core/trino-main/src/main/java/io/trino/execution/QueryState.java

```diff
@@ -73,4 +73,23 @@ public boolean isDone()
     {
         return doneState;
     }
+
+    public boolean isPlanningFinished()
+    {
+        switch (this) {
+            case QUEUED:
+            case WAITING_FOR_RESOURCES:
+            case DISPATCHING:
+            case PLANNING:
+                return false;
+            case STARTING:
+            case RUNNING:
+            case FINISHING:
+            case FINISHED:
+            case FAILED:
+                return true;
+            default:
```

Comment:
https://github.com/trinodb/trino/blob/master/DEVELOPMENT.md#avoid-default-clause-in-exhaustive-enum-based-switch-statements